src/main/java/org/prebid/server/bidder/sspbc/SspbcBidder.java

```diff
@@ -0,0 +1,330 @@
+import com.iab.openrtb.request.Format;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.request.Site;
+import com.iab.openrtb.response.Bid;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.utils.URIBuilder;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderCall;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.bidder.sspbc.request.RequestType;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.sspbc.ExtImpSspbc;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Collectors;
+
+public class SspbcBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpSspbc>> SSPBC_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String ADAPTER_VERSION = "5.8";
+    private static final String IMP_FALLBACK_SIZE = "1x1";
+    private static final String PREBID_SERVER_INTEGRATION_TYPE = "4";
+    private static final String BANNER_TEMPLATE = """
+            <html><head><title></title><meta charset="UTF-8"><meta name="viewport"\
+             content="width=device-width, initial-scale=1.0"><style> body { background-color:\
+             transparent; margin: 0; padding: 0; }</style><script> window.rekid = {{.SiteId}};\
+             window.slot = {{.SlotId}}; window.adlabel = '{{.AdLabel}}'; window.pubid = '{{.PubId}}';\
+             window.wp_sn = 'sspbc_go'; window.page = '{{.Page}}'; window.ref = '{{.Referer}}';\
+             window.mcad = {{.McAd}}; window.inver = '{{.Inver}}'; </script></head><body><div id="c"></div><script\
+             async crossorigin nomodule src="//std.wpcdn.pl/wpjslib/wpjslib-inline.js"\
+             id="wpjslib"></script><script async crossorigin type="module"\
+             src="//std.wpcdn.pl/wpjslib6/wpjslib-inline.js" id="wpjslib6"></script></body></html>""";
+
+    private final Map<Imp, ExtImpSspbc> impToExt = new HashMap<>();
```

Comment:
Pls, don't use mutable state in adapters. It can cause concurrent issues. Move it to local variable.